lua/packer.lua

```diff
@@ -188,13 +291,19 @@ manage = function(plugin)
     end
   end
 
-  plugin.install_path = util.join_paths(plugin.opt and config.opt_dir or config.start_dir,
-                                        plugin.short_name)
+  if plugins[plugin.short_name] and config.merge_duplicates then
+    log.info("Merging specs for plugin " .. plugin.short_name)
+    merge_specs(plugins[plugin.short_name], plugin)
+  else
+    plugin.install_path = util.join_paths(plugin.opt and config.opt_dir or config.start_dir,
+      plugin.short_name)
+
+    if not plugin.type then plugin_utils.guess_type(plugin) end
+    if plugin.type ~= plugin_utils.custom_plugin_type then plugin_types[plugin.type].setup(plugin) end
+    for k, v in pairs(plugin) do if handlers[k] then handlers[k](plugins, plugin, v) end end
 
-  if not plugin.type then plugin_utils.guess_type(plugin) end
-  if plugin.type ~= plugin_utils.custom_plugin_type then plugin_types[plugin.type].setup(plugin) end
-  for k, v in pairs(plugin) do if handlers[k] then handlers[k](plugins, plugin, v) end end
-  plugins[plugin.short_name] = plugin
+    plugins[plugin.short_name] = plugin
+  end
   if plugin.rocks then use_rocks(plugin.rocks) end
 
   if plugin.requires and config.ensure_dependencies then
```

Comment:
This entire chunk maybe looks more complicated than it actually is—all I'm doing here is to ask, "has this already been `use`d before? If not, do \<whatever packer did before\>. If we've seen it before, skip the `setup` stuff (since it's already been done the first time around) and merge the new spec into the old one".